src/app/core/services/filters.service.ts

```diff
@@ -365,4 +378,92 @@ export class FiltersService {
     const milestones = this.milestoneService.milestones;
     return [...milestones, Milestone.PRWithoutMilestone, Milestone.IssueWithoutMilestone];
   }
+
+  private getGhFilterDeselectedLabels(deselectedLabels: Set<string>): string {
+    return Array.from(deselectedLabels)
+      .map((label) => BooleanConjunctions.EXCLUDE + FilterOptions.label + `\"${label}\"`)
+      .join(BooleanConjunctions.AND);
+  }
+
+  private getGhFilterLabels(labels: string[]): string {
+    return labels.map((label) => FilterOptions.label + `\"${label}\"`).join(BooleanConjunctions.AND);
+  }
+
+  private getGhFilterMilestones(milestones: string[]): string {
+    return milestones
+      .map((milestone) => (MilestoneFilter.hasOwnProperty(milestone) ? MilestoneFilter[milestone] : FilterOptions.milestone + milestone))
+      .join(BooleanConjunctions.OR);
+  }
+
+  private getGhFilterSort(sort: Sort): string {
+    return SortFilter.hasOwnProperty(sort.active) ? SortFilter[sort.active] + ':' + sort.direction : '';
+  }
+
+  private getGhFilterTypes(type: string): string {
+    return TypeFilter[type];
+  }
+
+  /**
+   * Returns the encoded filter string for the GitHub search using url queries
+   * @returns Encoded filter string
+   */
+  getEncodedFilter(): string {
+    const res = [
+      '',
+      this.getGhFilterDeselectedLabels(this.filter$.value.deselectedLabels),
+      this.getGhFilterLabels(this.filter$.value.labels),
+      this.getGhFilterMilestones(this.filter$.value.milestones),
+      this.getGhFilterSort(this.filter$.value.sort),
+      this.getGhFilterTypes(this.filter$.value.type),
+      this.getGhFilterOpenAndClosedPR(this.filter$.value.assignees, this.filter$.value.status),
+      this.filter$.value.title
+    ];
+
+    return res
+      .filter((curr) => curr !== '')
+      .map((curr) => '(' + curr + ')')
+      .join(BooleanConjunctions.AND);
+  }
+
+  private getGhFilterOpenAndClosedPR(assignees: string[], status: string[]): string {
+    const toState = (stat: string): string => {
+      switch (stat) {
+        case StatusOptions.OpenPullRequests:
+        case StatusOptions.OpenIssues:
+          return 'is:open';
+        case StatusOptions.MergedPullRequests:
+          return 'is:merged';
+        case StatusOptions.ClosedPullRequests:
+        case StatusOptions.ClosedIssues:
+          return 'is:closed';
+        default:
+          return '';
+      }
+    };
+
+    const isIssue = (stat: string): boolean => stat === StatusOptions.OpenIssues || stat === StatusOptions.ClosedIssues;
+
+    const prFilter = status.filter((stat) => !isIssue(stat)).map(toState);
+    const issueFilter = status.filter(isIssue).map(toState);
+
+    if (prFilter.length === 0 && prFilter.length === 0) {
```

Comment:
These 2 conditionals are the same so checking it once is enough.

```suggestion
    if (prFilter.length === 0) {
```